elements/pfe-primary-detail/src/pfe-primary-detail.js

```diff
@@ -0,0 +1,177 @@
+import PFElement from "../../pfelement/dist/pfelement.js";
+
+class PfePrimaryDetail extends PFElement {
+  static get tag() {
+    return "pfe-primary-detail";
+  }
+
+  static get meta() {
+    return {
+      title: "Primary detail",
+      description: ""
+    };
+  }
+
+  get templateUrl() {
+    return "pfe-primary-detail.html";
+  }
+
+  get styleUrl() {
+    return "pfe-primary-detail.scss";
+  }
+
+  // static get events() {
+  //   return {
+  //   };
+  // }
+
+  // Declare the type of this component
+  static get PfeType() {
+    return PFElement.PfeTypes.Container;
+  }
+
+  static get properties() {
+    return {};
+  }
+
+  constructor() {
+    super(PfePrimaryDetail, { type: PfePrimaryDetail.PfeType });
+
+    this._handleHideShow = this._handleHideShow.bind(this);
+
+    this._detailsNav = this.shadowRoot.getElementById("details-nav");
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    // If you need to initialize any attributes, do that here
+
+    this._slots = {
+      detailsNav: this.shadowRoot.querySelector('slot[name="details-nav"]').assignedElements(),
+      details: this.shadowRoot.querySelector('slot[name="details"]').assignedElements(),
+      detailsNavHeader: this.shadowRoot.querySelector('slot[name="details-nav--header"]').assignedElements(),
+      detailsNavFooter: this.shadowRoot.querySelector('slot[name="details-nav--footer"]').assignedElements()
+    };
+
+    this._detailsWrapper = this.shadowRoot.getElementById("details-wrapper");
+
+    // Add appropriate markup and behaviors
+    this._scanLightDom();
```

Comment:
You could optionally wrap this in a `if(this.hasLightDOM())` to ensure light DOM exists before running the scan.

Reply:
Uh oh, is this a problem we run into? I assumed connectedCallback was always run when we have lightDom?

Reply:
And if I wrap in an if, I'd have to make sure it gets run when lightDom exists, right?

Reply:
connectedCallback is run if the tag is put on the page, but if the tag doesn't have any content, you probably don't want to parse it yet.  Maybe check if light DOM exists and if it doesn't, go ahead and attach the mutation observer?

Reply:
Ahh, I understand now, thanks!